src/core-concepts-in-embedded-application/logging.md

```diff
@@ -0,0 +1,118 @@
+# Logging
+
+​Logging is a crucial aspect of embedded systems development, providing visibility into the system's behavior and aiding in debugging and monitoring. In the Rust ecosystem for ESP devices, two prominent logging frameworks are commonly used: the [log] crate and [defmt]. The [esp-println] crate serves as a bridge, facilitating logging through these frameworks on ESP devices.​
+
+## The `log` crate
+
+The [log] crate is a widely adopted logging facade in the Rust community. It defines a set of macros (`info!`, `warn!`, `error!`, etc.) to capture log messages at various levels. The actual logging behavior is determined by the selected backend implementing the `Log` trait.
```

Comment:
> The actual logging behavior is determined by the selected backend implementing the `Log` trait.

This is such a niche implementation detail that it probably shouldn't be in an introductory paragraph.